plugin/pkg/bastion/validator/admission.go

```diff
@@ -151,7 +157,7 @@ func (v *Bastion) Admit(ctx context.Context, a admission.Attributes, _ admission
 	}
 
 	// ensure shoot SSH access is not disabled
-	if bastion.DeletionTimestamp == nil && !gardencorehelper.ShootEnablesSSHAccess(shoot) {
+	if bastion.DeletionTimestamp == nil && !gardencorehelper.ShootEnablesSSHAccess(&coreShoot) {
```

Comment:
If this is the only place `coreShoot` is used, we don't need to convert the object. You can call `v1beta1helper.ShootEnablesSSHAccess`